### ISSUE TITLE
wp-env: docker-compose fails on PHP < 7.2 due to Xdebug

### ISSUE THREAD

**Issue by swissspidy** [WordPress/gutenberg#27783]
**Describe the bug**

This is a follow-up to #27346 (cc @noahtallen).

With the new 3.0.0 version, I'm getting the following error when running `WP_ENV_PHP_VERSION =5.6 wp-env start`:

```
Service 'wordpress' failed to build : The command '/bin/sh -c apt -qy install $PHPIZE_DEPS 	&& pecl install xdebug 	&& docker-php-ext-enable xdebug' returned a non-zero code: 1
Step 2/6 : RUN apt -qy install $PHPIZE_DEPS 	&& pecl install xdebug 	&& docker-php-ext-enable xdebug

WARNING: channel "pecl.php.net" has updated its protocols, use "pecl channel-update pecl.php.net" to update
pecl/xdebug requires PHP (version >= 7.2.0, version <= 8.0.99), installed version is 5.6.39
No valid packages found
install failed
```

Note that I did not enable Xdebug. But I guess it's installed either way to make things easier.

I suppose the right version of Xdebug would need to be specified when running `pecl install`.

**To reproduce**

Run `WP_ENV_PHP_VERSION=5.6 wp-env start`

**Expected behavior**

Running the container on 5.6 should work just fine like it used to.

**Comment by noahtallen** [comment]
Good catch! I think my original intention was to only support Xdebug on the latest PHP version. Otherwise, if we have to support multiple versions of Xdebug, we can’t really share a configuration pattern. For example, between Xdebug 2 and 3, the configuration changed pretty drastically. So if we need to enable Xdebug 2 only for older PHP versions, we then have to change how we configure it depending on the version, which starts getting messy really quickly.

**Comment by swissspidy** [comment]
Only supporting Xdebug 3, which works on PHP 7.2+, seems fair. Not installing it on older versions of PHP at all in that case.

**Comment by davidshq** [comment]
I'm getting this error running `wp-env start` without any params:

```
Service 'wordpress' failed to build: The command '/bin/sh -c apt -qy install $PHPIZE_DEPS       && pecl install xdebug  && docker-php-ext-enable xdebug' returned a non-zero code: 1
```

**Comment by ZebulanStanphill** [comment]
Not sure if this is the same issue or not, but I've run into something quite similar working on a private project. Here's my `.wp-env.json`:

```json
{
	"core": null,
	"phpVersion": "7.4",
	"config": {
		"WP_DEBUG": true,
		"WP_DEBUG_DISPLAY": true
	},
	"plugins": ["."]
}
```

When I try to run `npm run wp-env start`, it results in the following error:

```
✖ Error while running docker-compose command.
Step 1/6 : FROM wordpress:php7.4
 ---> 39fc6a7fa0c2
Step 2/6 : RUN apt -qy install $PHPIZE_DEPS     && pecl install xdebug  && docker-php-ext-enable xdebug
 ---> Running in f37269464a15

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Reading package lists...
Building dependency tree...
Reading state information...
autoconf is already the newest version (2.69-11).
dpkg-dev is already the newest version (1.19.7).
file is already the newest version (1:5.35-4+deb10u2).
g++ is already the newest version (4:8.3.0-1).
gcc is already the newest version (4:8.3.0-1).
libc6-dev is already the newest version (2.28-10).
make is already the newest version (4.2.1-1.2).
pkg-config is already the newest version (0.29-6).
re2c is already the newest version (1.1.1-1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
pecl/xdebug is already installed and is the same as the released version 3.0.3
install failed
Building wordpress
Service 'wordpress' failed to build: The command '/bin/sh -c apt -qy install $PHPIZE_DEPS       && pecl install xdebug  && docker-php-ext-enable xdebug' returned a non-zero code: 1
```

If I remove the `plugins` field from the config file, it works fine.

**Comment by noahtallen** [comment]
@ZebulanStanphill I think the warning part of that message was fixed by https://github.com/WordPress/gutenberg/pull/28840 -- could you verify if you still get a different error there?

**Comment by ZebulanStanphill** [comment]
@noahtallen I now get a different error, so progress has definitely been made:

```
✖ Error while running docker-compose command.
chown: cannot access 'wp-config.php': No such file or directory
```

**Comment by noahtallen** [comment]
> @noahtallen I now get a different error, so progress has definitely been made:

Interesting! That was supposed to be solved by #30053 🤔
